### PR TITLE
`Pipeline`: Use capabilities instead of hardcoded LLMs

### PR DESCRIPTION
**Pull request by kaancayli**
### Motivation

In the future, we want Pyris to use a capability based system. This PR enables the use of the existing capability system in existing pipelines. Other than that, MessageDTO and IrisMessage models are replaced by PyrisMessage class.

## Summary by CodeRabbit

- **New Features**
	- Introduced `PyrisMessage` model and `IrisMessageRole` enumeration for enhanced message handling.
	- Updated message conversion functionality, including role mapping and content checking.
- **Enhancements**
	- Improved chat functionality in tutor chat pipeline with new message types and better exception handling.
	- Enhanced message format conversion in external communication modules.
- **Refactor**
	- Updated import paths across various modules for better code organization.
	- Renamed `IrisMessage` to `PyrisMessage` in multiple files for consistency.
- **Bug Fixes**
	- Adjusted optional attribute types in DTOs to handle null values more gracefully.

### CODE REVIEW

File: app/llm/request_handler/request_handler_interface.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from abc import ABCMeta, abstractmethod
 
-from ...domain import IrisMessage
+from ...domain import PyrisMessage
```
[the remark this answers is not in the document]
Comment (kaancayli) [reply app/llm/request_handler/request_handler_interface.py:3]: There are no references remaining to old IrisMessage. You can verify it by running:
#!/bin/bash
# Verify that all instances of `IrisMessage` are replaced with `PyrisMessage`.
rg --type py 'IrisMessage$'

**Comment by kaancayli** [comment]
@Hialus I have updated the problematic import statements. For general overhaul of import statements I will create an issue

File: app/pipeline/chat/file_selector_pipeline.py [hunk 2]
```diff
@@ -41,7 +42,19 @@ class FileSelectorPipeline(Pipeline):
...
+                input_cost=1,
+                output_cost=1,
+                gpt_version_equivalent=3.5,
+                context_length=4096,
+                vendor="OpenAI",
+                privacy_compliance=False,
+                self_hosted=False,
+                image_recognition=False,
+                json_mode=False,
+            )
```
Comment (Hialus) [inline app/pipeline/chat/file_selector_pipeline.py:56]: You do not have to specify every single requirement here. You should only specify those that are important. Here it should be enough to specify the context length and the gpt version

File: app/pipeline/chat/tutor_chat_pipeline.py [hunk 3]
```diff
@@ -43,7 +45,19 @@ class TutorChatPipeline(Pipeline):
...
+                output_cost=1,
+                gpt_version_equivalent=3.5,
+                context_length=4096,
+                vendor="OpenAI",
+                privacy_compliance=False,
+                self_hosted=False,
+                image_recognition=False,
+                json_mode=False,
+            )
+        )
```
Comment (Hialus) [inline app/pipeline/chat/tutor_chat_pipeline.py:60]: Same as above, however this one has to also be privacy compliant, as it handles user data.

**Review by Hialus — CHANGES_REQUESTED** [review]
Looks mostly good, however the capability declarations are not sensible.

Also you did not adjust the SummaryPipeline.

File: app/pipeline/chat/tutor_chat_pipeline.py (hunk 3 above)
Comment (kaancayli) [reply app/pipeline/chat/tutor_chat_pipeline.py]: Again thanks for the hint. Should be resolved

File: app/pipeline/chat/file_selector_pipeline.py (hunk 2 above)
Comment (kaancayli) [reply app/pipeline/chat/file_selector_pipeline.py]: Thanks for the hint. Addressed it now.

**Pull request merged**